lib/main.dart

```diff
@@ -13,8 +13,8 @@ void main() {
     return true;
   }());
   LicenseRegistry.addLicense(additionalLicenses);
-  LiveZulipBinding.ensureInitialized();
   WidgetsFlutterBinding.ensureInitialized();
+  LiveZulipBinding.ensureInitialized();
   NotificationService.instance.start();
```

Comment:
Yes they need to be reordered, since `deviceInfo` is being prefetched in `LiveZulipBinding.ensureInitialized()` and `PlatformChannel`s require `WidgetsFlutterBinding` to be initialized first.

Reply:
What does the error look like if the order of these gets swapped back?

If it doesn't make clear that the problem is that `WidgetsFlutterBinding` is needed first, then ideally we can do something to make the error clear.

If that's hard or annoying, then just a comment here explaining this ordering constraint would also be an acceptable solution.

Reply:
Here's how the errors look like:
```
flutter: Failed to prefetch device info: Binding has not yet been initialized.
The "instance" getter on the ServicesBinding binding mixin is only available once that binding has been initialized.
Typically, this is done by calling "WidgetsFlutterBinding.ensureInitialized()" or "runApp()" (the latter calls the former). Typically this call is done in the "void main()" method. The "ensureInitialized" method is idempotent; calling it multiple times is not harmful. After calling that method, the "instance" getter will return the binding.
In a test, one can call "TestWidgetsFlutterBinding.ensureInitialized()" as the first line in the test's "main()" method to initialize the binding.
If ServicesBinding is a custom binding mixin, there must also be a custom binding class, like WidgetsFlutterBinding, but that mixes in the selected binding, and that is the class that must be constructed before using the "instance" getter.
#0      BindingBase.checkInstance.<anonymous closure> (package:flutter/src/foundation/binding.dart:309:9)
#1      BindingBase.checkInstance (package:flutter/src/foundation/binding.dart:390:6)
#2      ServicesBinding.instance (package:flutter/src/services/binding.dart:68:54)
#3      _findBinaryMessenger (package:flutter/src/services/platform_channel.dart:158:25)
#4      MethodChannel.binaryMessenger (package:flutter/src/services/platform_channel.dart:293:56)
#5      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:327:15)
#6      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:507:12)
#7      MethodChannelDeviceInfo.deviceInfo (package:device_info_plus_platform_interface/method_channel/method_channel_device_info.dart:19:24)
#8      DeviceInfoPlugin.macOsInfo (package:device_info_plus/device_info_plus.dart:86:48)
#9      DeviceInfoPlugin.deviceInfo (package:device_info_plus/device_info_plus.dart:107:16)
#10     LiveZulipBinding._prefetchDeviceInfo (package:zulip/model/binding.dart:338:62)
#11     new LiveZulipBinding (package:zulip/model/binding.dart:290:19)
#12     LiveZulipBinding.ensureInitialized (package:zulip/model/binding.dart:297:7)
#13     main (package:zulip/main.dart:16:20)
#14     _runMain.<anonymous closure> (dart:ui/hooks.dart:301:23)
#15     _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:297:19)
#16     _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)

flutter: Failed to prefetch package info: Binding has not yet been initialized.
The "instance" getter on the ServicesBinding binding mixin is only available once that binding has been initialized.
Typically, this is done by calling "WidgetsFlutterBinding.ensureInitialized()" or "runApp()" (the latter calls the former). Typically this call is done in the "void main()" method. The "ensureInitialized" method is idempotent; calling it multiple times is not harmful. After calling that method, the "instance" getter will return the binding.
In a test, one can call "TestWidgetsFlutterBinding.ensureInitialized()" as the first line in the test's "main()" method to initialize the binding.
If ServicesBinding is a custom binding mixin, there must also be a custom binding class, like WidgetsFlutterBinding, but that mixes in the selected binding, and that is the class that must be constructed before using the "instance" getter.
#0      BindingBase.checkInstance.<anonymous closure> (package:flutter/src/foundation/binding.dart:309:9)
#1      BindingBase.checkInstance (package:flutter/src/foundation/binding.dart:390:6)
#2      ServicesBinding.instance (package:flutter/src/services/binding.dart:68:54)
#3      _findBinaryMessenger (package:flutter/src/services/platform_channel.dart:158:25)
#4      MethodChannel.binaryMessenger (package:flutter/src/services/platform_channel.dart:293:56)
#5      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:327:15)
#6      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:507:12)
#7      MethodChannel.invokeMapMethod (package:flutter/src/services/platform_channel.dart:534:49)
#8      MethodChannelPackageInfo.getAll (package:package_info_plus_platform_interface/method_channel_package_info.dart:13:32)
#9      PackageInfo.fromPlatform (package:package_info_plus/package_info_plus.dart:80:61)
#10     LiveZulipBinding._prefetchPackageInfo (package:zulip/model/binding.dart:367:56)
#11     new LiveZulipBinding (package:zulip/model/binding.dart:291:20)
#12     LiveZulipBinding.ensureInitialized (package:zulip/model/binding.dart:297:7)
#13     main (package:zulip/main.dart:16:20)
#14     _runMain.<anonymous closure> (dart:ui/hooks.dart:301:23)
#15     _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:297:19)
#16     _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)
```

Basically both of the prefetch fail because of same reason — MethodChannel requiring `WidgetsFlutterBinding` initialized first.

Reply:
Cool, that seems clear enough — thanks.